Automattic/wp-calypso
# Subscription Management: Pass email_id url param to unsubscribe API endpoint on individual site subscription page.

Fixes https://github.com/Automattic/wp-calypso/issues/78293

## Proposed Changes

This PR makes the unsubscribe action in the Individual Subscription page pass email_id to the unsubscribe endpoint if that parameter is on the URL.
<img width="843" alt="Screenshot 2023-06-15 at 18 47 38" src="https://github.com/Automattic/wp-calypso/assets/3832570/c25590a4-f8b5-4a2b-89ab-ab126614c71c">

## Testing Instructions
Note: use these instructions for internal and external users.

For internal users the endpoint should have a structure like this:
<img width="522" alt="Screenshot 2023-06-15 at 18 38 06" src="https://github.com/Automattic/wp-calypso/assets/3832570/e8c444c9-40c2-4e04-b85f-f1032641b78b">

For external users the endpoint should have a structure like this:
<img width="323" alt="Screenshot 2023-06-15 at 18 39 53" src="https://github.com/Automattic/wp-calypso/assets/3832570/b565a89b-0001-4ba5-8455-f6b12f6f6d5a">

1. Apply this patch and start the application.
2. Go to `http://calypso.localhost:3000/subscriptions/site/{id of any page you are subscribed to}`.
3. Open the network tab in the developer tools.
4. Click on `Cancel subscription` and confirm the cancellation.
5. Check the URL of the unsubscribe endpoint. It shouldn't contain any reference to `email_id` on it.
6. Resubscribe using the resubscribe button on the body of the page.
7. Add the `email_id` parameter with any number, so the URL is like this: `http://calypso.localhost:3000/subscriptions/site/{id of any page you are subscribed to}?email_id={put any number you want here}`.
8. Click on `Cancel subscription` and confirm the cancellation.
9. Check the URL of the unsubscribe endpoint. It should contain a new query parameter `email_id` on it.
